python/tidy/servo_tidy/tidy.py

```diff
@@ -327,28 +326,25 @@ def check_by_line(file_name, lines):
 
 
 def check_flake8(file_name, contents):
-    from flake8.main import check_code
-
     if not file_name.endswith(".py"):
         raise StopIteration
 
-    @contextlib.contextmanager
-    def stdout_redirect(where):
-        sys.stdout = where
-        try:
-            yield where
-        finally:
-            sys.stdout = sys.__stdout__
+    temp_file = tempfile.NamedTemporaryFile()
+    temp_file.write(contents)
+    temp_file.flush()
```

Comment:
Why create this temporary file rather than pass `file_name` to flake8?

Reply:
I wondered that myself and choose to dump 
the content because I did not take time to figure out if file_name really existed on disk

Reply:
Should I change that before merging?

Reply:
Yes, please do.

Reply:
Done!